### DISCUSSION TITLE
Particle Removal

### DISCUSSION THREAD

**Discussion by CDaviesAFF** [firemodels/fds#13630] [Q&A] [answered]
Hello, 

While working on a larger project I noticed a large pooling effect due to the water particles not being removed from the model and eventually causing the model to crash due to excessive load on the processor.

To attempt to figure this out I have created a simplified model to test this; simply a 20x20m box with a sprinkler head.

&HEAD CHID='Sprinkler_Test'/
&TIME T_END=300.0/
&DUMP DT_RESTART=300.0, DT_SL3D=0.25/

&MESH ID='Mesh01', IJK=110,110,30, XB=-1.0,21.0,-1.0,21.0,0.0,6.0/

&SPEC ID='WATER VAPOR'/

&PART ID='Water',
      SPEC_ID='WATER VAPOR',
      DIAMETER=5636.0,
      AGE=60.0,
      SAMPLING_FACTOR=1/

&PROP ID='Generic Commercial Spray',
      PART_ID='Water',
      OFFSET=0.24,
      FLOW_RATE=60.0,
      PARTICLE_VELOCITY=7.5,
      SPRAY_ANGLE=0.0,55.0/
&DEVC ID='SPRK', PROP_ID='Generic Commercial Spray', XYZ=10.0,10.0,4.8, QUANTITY='TIME', SETPOINT=10.0/

&OBST ID='Obstruction', XB=-1.0,10.0,-1.0,20.0,2.8,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,21.0,5.0,5.2, COLOR='INVISIBLE'/ 
&OBST ID='Obstruction', XB=-1.0,-5.551115E-17,-5.551115E-17,20.0,0.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,-5.551115E-17,0.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,20.0,21.0,0.0,5.0/ 
&OBST ID='Obstruction', XB=20.0,21.0,-5.551115E-17,20.0,0.0,5.0/ 

&VENT ID='Mesh Vent: Mesh01 [XMAX]', SURF_ID='OPEN', XB=21.0,21.0,-1.0,21.0,0.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,21.0,0.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01 [YMAX]', SURF_ID='OPEN', XB=-1.0,21.0,21.0,21.0,0.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01 [YMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,-1.0,0.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01 [ZMAX]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,21.0,6.0,6.0/ 

&TAIL /
![image](https://github.com/user-attachments/assets/708dc6cc-893d-46c0-a706-241171076f4c)

Here the pooling effect can be seen on the obstruction and the particles are removed as per expected at z = 0.
my question is with the following model, the only change being splitting the mesh @ z=3 (where the floor slab obstruction is placed). in this model the particles are removed at both z = 3 and z = 0. 

&HEAD CHID='Sprinkler_Test_2_mesh'/
&TIME T_END=300.0/
&DUMP DT_RESTART=300.0, DT_SL3D=0.25/

&MESH ID='Mesh01-01', IJK=110,110,15, XB=-1.0,21.0,-1.0,21.0,0.0,3.0/
&MESH ID='Mesh01-02', IJK=110,110,15, XB=-1.0,21.0,-1.0,21.0,3.0,6.0/

&SPEC ID='WATER VAPOR'/

&PART ID='Water',
      SPEC_ID='WATER VAPOR',
      DIAMETER=5636.0,
      AGE=60.0,
      SAMPLING_FACTOR=1/

&PROP ID='Generic Commercial Spray',
      PART_ID='Water',
      OFFSET=0.24,
      FLOW_RATE=60.0,
      PARTICLE_VELOCITY=7.5,
      SPRAY_ANGLE=0.0,55.0/
&DEVC ID='SPRK', PROP_ID='Generic Commercial Spray', XYZ=10.0,10.0,4.8, QUANTITY='TIME', SETPOINT=10.0/

&OBST ID='Obstruction', XB=-1.0,10.0,-1.0,20.0,2.8,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,21.0,5.0,5.2, COLOR='INVISIBLE'/ 
&OBST ID='Obstruction', XB=-1.0,-5.551115E-17,-5.551115E-17,20.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,-5.551115E-17,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,20.0,21.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=20.0,21.0,-5.551115E-17,20.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,-5.551115E-17,-5.551115E-17,20.0,3.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,-5.551115E-17,3.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,20.0,21.0,3.0,5.0/ 
&OBST ID='Obstruction', XB=20.0,21.0,-5.551115E-17,20.0,3.0,5.0/ 

&VENT ID='Mesh Vent: Mesh01-01 [XMAX]', SURF_ID='OPEN', XB=21.0,21.0,-1.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [YMAX]', SURF_ID='OPEN', XB=-1.0,21.0,21.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [YMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,-1.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [ZMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,21.0,0.0,0.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [XMAX]', SURF_ID='OPEN', XB=21.0,21.0,-1.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [YMAX]', SURF_ID='OPEN', XB=-1.0,21.0,21.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [YMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,-1.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [ZMAX]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,21.0,6.0,6.0/ 

&TAIL /

![image](https://github.com/user-attachments/assets/7d25136d-01c6-479e-83fb-03968674de1a)

My question is why does the model remove the particles in the second instance and not the first and the intent behind my question is to do with the cooling effect a layer of water would have on a larger model with a fire included. I understand it is possible to remove the water particles using a vent as per 15.6 in the user guide. But I would like to understand better how FDS operates so to be able to more accurately build model in the future. (and to avoid large cpu loads due to the amount of particles present)

A secondary question but far less important to me (simply something I noticed while copying across the fds file) why does the FDS file split the obstruction coordinates up between the meshes (i assume this is to assist in processing each mesh)

**Comment by CDaviesAFF** [comment]
Is there something in the background of FDS that is changing the state of POROUS_FLOOR between these models that I am not understanding?

**Comment by CDaviesAFF** [comment]
Further to this I have modelled the same thing but with the "floor slab" 1m lower i.e. not at the mesh boundary and the pooling effect has returned

![image](https://github.com/user-attachments/assets/0ba320e8-bbd8-4176-ab10-a67c8fcfb2ff)

&HEAD CHID='Sprinkler_Test_2_mesh_lower_floor'/
&TIME T_END=300.0/
&DUMP DT_RESTART=300.0, DT_SL3D=0.25/

&MESH ID='Mesh01-01', IJK=110,110,15, XB=-1.0,21.0,-1.0,21.0,0.0,3.0/
&MESH ID='Mesh01-02', IJK=110,110,15, XB=-1.0,21.0,-1.0,21.0,3.0,6.0/

&SPEC ID='WATER VAPOR'/

&PART ID='Water',
      SPEC_ID='WATER VAPOR',
      DIAMETER=5636.0,
      AGE=60.0,
      SAMPLING_FACTOR=1/

&PROP ID='Generic Commercial Spray',
      PART_ID='Water',
      OFFSET=0.24,
      FLOW_RATE=60.0,
      PARTICLE_VELOCITY=7.5,
      SPRAY_ANGLE=0.0,55.0/
&DEVC ID='SPRK', PROP_ID='Generic Commercial Spray', XYZ=10.0,10.0,4.8, QUANTITY='TIME', SETPOINT=10.0/


&OBST ID='Obstruction', XB=-1.0,10.0,-1.0,21.0,1.8,2.0/ 
&OBST ID='Obstruction', XB=-1.0,-5.551115E-17,-5.551115E-17,20.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,-5.551115E-17,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,20.0,21.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=20.0,21.0,-5.551115E-17,20.0,0.0,3.0/ 
&OBST ID='Obstruction', XB=-1.0,-5.551115E-17,-5.551115E-17,20.0,3.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,-1.0,-5.551115E-17,3.0,5.0/ 
&OBST ID='Obstruction', XB=-1.0,21.0,20.0,21.0,3.0,5.0/ 
&OBST ID='Obstruction', XB=20.0,21.0,-5.551115E-17,20.0,3.0,5.0/ 

&VENT ID='Mesh Vent: Mesh01-01 [XMAX]', SURF_ID='OPEN', XB=21.0,21.0,-1.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [YMAX]', SURF_ID='OPEN', XB=-1.0,21.0,21.0,21.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-01 [YMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,-1.0,0.0,3.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [XMAX]', SURF_ID='OPEN', XB=21.0,21.0,-1.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [XMIN]', SURF_ID='OPEN', XB=-1.0,-1.0,-1.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [YMAX]', SURF_ID='OPEN', XB=-1.0,21.0,21.0,21.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [YMIN]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,-1.0,3.0,6.0/ 
&VENT ID='Mesh Vent: Mesh01-02 [ZMAX]', SURF_ID='OPEN', XB=-1.0,21.0,-1.0,21.0,6.0,6.0/ 


&TAIL /

I am curious why FDS has a different approach when the obstruction is placed on the mesh boundary versus not. Again this is simply me attempting to better understand FDS

**Comment by drjfloyd** [comment]
The porous floor only removes particles if there is no OBST ontop of the lowest boundary of the domain. You can also limit memory useage by setting AGE to a lower value and/or MAXIMUM_PARTICLES to a lower value.

**Reply by CDaviesAFF** [reply]
Hello Dr Floyd,

Thank you for your response. I am aware of the particle duration and maximum particle functions but my question was more to the point of how FDS calculates as per the wording in the User Guide chapter 15.6 "the 'floor' of the computational domain" as the particles from the sprinkler are removed by the obst on the mesh boundary despite their being 3m of computational domain i.e. mesh below the obst

This is considering the particle age was still the default 60s (the particles only remained within the model for the ~1 sec it took for them to reach the obst) and there was no extracting vent to remove them the only option remaining is that they must have reached the floor of the computational domain.

I would love any more clarifying information around this issue.

Thanks,

**Comment by drjfloyd** [comment, marked as answer]
I think there is a bug with the OBST being exactly at a mesh boundary. If I remove the OBST at Z=2.8 to 3.0, particles fall from one mesh to next.  Going to convert this to an issue.
